ideascube/static/ideascube/js/ideascube.js

```diff
@@ -265,3 +265,26 @@ ID.confirmClick = function (selector) {
     };
     el.addEventListener('click', ask, false);
 };
+
+
+ID.initWifiList = function(item_selector, popup_selector, urlroot) {
+    var elements = document.querySelectorAll(item_selector);
+
+    for (var element of elements) {
+        var known = element.getAttribute('data-known') === 'True';
+        var secure = element.getAttribute('data-secure') === 'True';
+
+        if (known || !secure) {
+            var ssid = element.getAttribute('data-ssid');
+            element.setAttribute('href', urlroot + ssid);
+        } else {
+            element.setAttribute('href', popup_selector);
+
+            element.addEventListener('click', function(evt) {
```

Comment:
The popup is only displayed in some cases (the `else` case), and it seems very dependent on the list, its items, and what's inside.

I'm not sure that could be reused directly anywhere else.

In general, I'm a bit cautious about making reusable functions before there's actually a place where it would be nice to reuse it, because only when you have that need can you really see how you'd want to reuse it. (i.e the needed API)

Can't we wait for having a need to display other popups, and at that point we can change that code, make it reusable, and reuse it?